### DISCUSSION TITLE
Spot an issue with documentation?  Report it here.

### DISCUSSION THREAD

**Discussion by randypacker** [dreamworksanimation/openmoonray#47] [Announcements]
Thanks for pointing out issues in our documentation!  We'd kindly ask to report any issues with documentation in this thread, as it's easier to take care of outside of the CLA / pull request process.

For documentation fixes in the repos, we'll address and they'll be updated in our releases.  For documentation on the website, we'll address and they should be fixed pretty quickly outside of the release process.  

The fixes to the README file will be in our next release.  That addresses #5 , #17, #16, #36 and #39

**Comment by alhirzel** [comment]
#3 too?

**Comment by xXTR0YXx** [comment]
The path to the white papers regarding path-guiding is dead:
https://docs.openmoonray.org/user-reference/how-to-guides/path-guiding/

use this here instead:
https://jannovak.info/publications/PathGuide/index.html

**Comment by knightcrawler25** [comment]
The [docs](https://docs.openmoonray.org/user-reference/tools/moonray/#command-line-options) for the moonray command line tool list the option `-texturesystem <sony/dwaproduction>` which does not appear to be referenced within the code nor does it show up when using `moonray -h`. Maybe this option is valid for an internal build of MoonRay but not for the open source release?

I assume the documentation is a work in progress because there are some [missing](https://docs.openmoonray.org/developer-reference/shaders/materials/)_[sections](https://docs.openmoonray.org/user-reference/scene-objects/volumes/) and some [broken links](https://docs.openmoonray.org/user-reference/scene-objects/materials/linkToShadingAPI?). Perhaps this is already known but wanted to point it out just in case it wasn't.

Lastly, not an issue but a rather tall order, especially given how big of a challenge it is for the internal team to document everything:
For the User Reference, it would be nice to have some visual examples of what the parameters do. The documentation for some of the [Display Filters](https://docs.openmoonray.org/user-reference/scene-objects/display-filters/ColorCorrectDisplayFilter/) already do this, so it would be great to have it for the other scene objects, wherever applicable.

Would it make sense for the documentation to be on GitHub so that the community can help?

**Reply by scegielski** [reply]
Not sure about the texturesystem but what you say sounds correct that it's for our internal build.   As for the documentation, thanks for pointing out the missing sections and broken links.   It's in progress and we typically work on and update it on Mondays.   We already have some visual examples as you point out and we are working on adding more.

**Reply by knightcrawler25** [reply]
Thank you, this helps :)

**Reply by jlanz** [reply]
Thanks - I think these issues have now been addressed.

**Comment by junwha** [comment]
[Contributionhttps://github.com/dreamworksanimation/openmoonray/blob/release/CONTRIBUTING.md](https://github.com/dreamworksanimation/openmoonray/blob/release/CONTRIBUTING.md)
Contributing document has a invalid url for the coding standards
Need to change
https://docs.openmoonray.org/developer-guide/coding-standards/
to
https://docs.openmoonray.org/developer-reference/coding-standards/

**Reply by scegielski** [reply]
Thanks, noted.

**Reply by jlanz** [reply]
Fixed!

**Comment by DJtheRedstoner** [comment]
https://docs.openmoonray.org/user-reference/tools/hydra/hdmoonray-setup/ uses the line `export PXR_PLUGINPATH_NAME=$REL/plugin/pxr/usd:${PXR_PLUGINPATH_NAME}` but the `$REL/plugin/pxr/usd` directory doesn't exist, it should be `$REL/plugin/usd`.

**Reply by rgwilson68** [reply]
Thanks for reporting this. It is now fixed.

**Comment by mpjohn** [comment]
This might be a silly question but I would love to see something in the docs about...

(a) animation - how to render out an animation instead of a still
(b) Options for assigning moonray shaders. Is it possible to do this outside of Houdini for instance?

**Reply by scegielski** [reply]
Hi,
As Moonray is primarily intended as a renderer, there are no inherent animation features per se.    It is typically up the the DCC (i.e. Houdini) to facilitate animation which can then be rendered using the Hydra delegate or by writing a series of rdl files.   That said, rdl files are written in the Lua language and therefore simple animation is possible using expressions.   For example, you could animate a square mesh across the frame like this:
```lua
SceneVariables {
    ["image_width"] = 500,
    ["image_height"] = 500,
}

PerspectiveCamera("camera") {
    ["node_xform"] = translate(0, 0, 10),
}

-- Animation from -5 to 5 over 30 frames
tx = -5.0 + (frame / 30.0) * 10.0

geom = RdlMeshGeometry("geom") {
    ["node_xform"] = translate(tx, 0, 0),
    ["face_vertex_count"] = { 4 },
    ["vertices_by_index"] = { 0, 1, 2, 3 },
    ["vertex_list_0"] = {
        Vec3(-1.0, -1.0, 0.0),
        Vec3(-1.0,  1.0, 0.0),
        Vec3( 1.0,  1.0, 0.0),
        Vec3( 1.0, -1.0, 0.0)
    }
}

envlight = EnvLight("envlight") { }

lightSet = LightSet("lightset") { envlight }

GeometrySet("geometryset") { geom }

material = UsdPreviewSurface("material") { }

Layer("layer") {
    { geom, {""}, material, lightSet }
}
```
You would then need to set the `frame` variable when rendering with moonray using the `-rdla_set` flag.   Rendering an animation  out to a series of .exr files could be accomplished with a simple bash script like this:
```bash
!#/bin/bash
for i in $(seq 1 30);
do
    moonray -in animation.rdla -out animation.$i.exr -rdla_set frame $i
done
```
If you wanted to add motion blur, this can be done using the `blur` command on the `node_xform` parameter.   See the following page:
https://docs.openmoonray.org/user-reference/how-to-guides/motion-blur

For assigning materials, please check out the page on the rdl *Layer* scene object here:
https://docs.openmoonray.org/user-reference/scene-objects/layer/Layer

Hope this helps.   Cheers.

**Reply by mpjohn** [reply]
Thanks Scegielski. That helps a lot. It's a shame that there isn't a Maya link but it looks like the only option is to figure out how to get it working inside Houdini ( easier said than done! )

**Reply by scegielski** [reply]
The Hydra delegate should also work with Maya but this hasn't been tested.

**Comment by mpjohn** [comment]
The current docs allow you to build Moonray....which is great up to a point.
It's a bit like being handed a new car...without wheels.
It's a lovely car. You can have a look around ( moonray_gui ) - but how do you go anywhere without a link to a DCC app?
Without reliable and reproducable instructions to build against Houdini or Maya we can't go anywhere.
Is there any plan to address this anytime soon?

**Comment by mpjohn** [comment]
If you were to set up Rez instructions for everyone outside of Dreamworks that could be very useful. 

A single set of clear instructions that work for integration into Houdini or Maya is vital to moving this thing forward. That is...if you want people to actually use Moonray rather than just to build it and have it sitting there because it can't be used with DCC apps.

**Comment by heavyrain266** [comment]
Are accssibility issues fine to be reported here? noticed a bug in the frontend which is leading to flash in form of blank, white page every time you open or refresh the page which is leading to issues related to epilepsy and seizures. Seems to be recent, didn't noticed it ealier.
